node/src/components/linear_chain_sync/operations.rs

```diff
@@ -455,7 +493,7 @@ pub(crate) async fn run_fast_sync_task(
         .era_start
         .saturating_diff(current_header.timestamp())
         < chainspec.deploy_config.max_ttl
-        && !current_header.is_genesis_child()
+        && current_header.height() != 0
```

Comment:
What if we're just at the beginning of an era/shortly after the key block - we might miss some deploys as we will never go past the key block's era start timestamp. We will never download deploys that were included in blocks before that key block.

Reply:
@afck is the original author of this...

Reply:
I don't understand the question.
The idea is that the current era began at `trusted_key_block_info.era_start`, so any blocks in it that we might need to validate and check for replays will have a timestamp greater or equal than that. For replay detection in a block with timestamp `t`, all deploys in blocks with timestamps greater or equal than `t - max_ttl` need to be considered. So we keep downloading old blocks until they are older than `trusted_key_block_info.era_start - max_ttl`.

Reply:
Right, thanks. I confused myself with the usage of single key block throughout.